PVTPackage/source/MultiphaseSystem/DeadOilMultiphaseSystem.cpp

```diff
@@ -62,15 +92,43 @@ std::unique_ptr< DeadOilMultiphaseSystem > DeadOilMultiphaseSystem::build( const
                                                                            const std::vector< double > & surfaceDensities,
                                                                            const std::vector< double > & molarWeights )
 {
+  const bool containsOil = std::find( phases.cbegin(), phases.cend(), pvt::PHASE_TYPE::OIL ) != phases.cend();
+  const bool containsGas = std::find( phases.cbegin(), phases.cend(), pvt::PHASE_TYPE::GAS ) != phases.cend();
+  const bool containsWater = std::find( phases.cbegin(), phases.cend(), pvt::PHASE_TYPE::LIQUID_WATER_RICH ) != phases.cend();  
+
+  const bool isValid = containsOil && ( containsGas || containsWater );
+  if( !isValid )  
+  {    
+    const std::string msg = "Three types of DO systems are allowed: Oil-Water-Gas, Oil-Water, and Oil-Gas";
+    LOGERROR( msg );
+  }
+   
   // props.oilTable, props.gasTable and props.waterTable respectively contain PVDO, PVDG and PVTW
```

Comment:
Here I would keep `oilTable`, `gasTable`, and `waterTable` because the `struct` `Properties` is used for both Black-Oil and Dead-Oil. For Dead-Oil the Intersect keywords are indeed `PVDO`, `PVDG`, `PVTW`, but for Black-Oil they are `PVTO`, `PVTG`, and `PVTW`. So with the current naming we can cover both cases.